### PR TITLE
feat: record/replay test harness with provider coverage

### LINKED ISSUE #89: test: git.rs and wt.rs provider tests with mock CommandRunner
## Context

- \`providers/vcs/git.rs\` (157 lines) — 0% coverage. Implements \`Vcs\` trait, shells out to \`git\`.
- \`providers/vcs/wt.rs\` (182 lines) — 0% coverage. Implements \`CheckoutManager\`, shells out to \`wt\`.

Both use \`run_cmd()\` for all external calls.

## Prerequisites

- #73 (CommandRunner trait)

### git.rs
- \`resolve_repo_root()\` — parses git rev-parse output
- \`branches()\` — parses branch list with ahead/behind info
- \`remote_branches()\` — filters remote branch output
- \`log()\` — parses git log output
- \`unpushed_count()\` — parses rev-list count
- \`working_tree_status()\` — parses porcelain status
- Error handling for each command failure

### wt.rs
- \`checkouts()\` — parses \`wt list --format=json\` output
- \`strip_to_json()\` — handles wt output format quirks
- \`switch()\` — correct args passed
- \`remove()\` — correct cleanup sequence
- Error handling

## Approach

With CommandRunner trait, provide canned stdout/stderr for each git/wt command. Test the parsing and error handling logic.

## Difficulty

**Medium-Hard** — depends on CommandRunner trait. Good coverage payoff once available.

### LINKED ISSUE #90: test: GitHub code_review and issue_tracker with mock GhApi
## Context

- \`code_review/github.rs\` (243 lines) — has 2 JSON parsing tests, but the async provider methods are untested
- \`issue_tracker/github.rs\` (98 lines) — has 1 PR-filtering test, async methods untested

Both inject \`Arc<GhApiClient>\` which can be mocked once it's a trait.

## Prerequisites

- #74 (GhApiClient trait)

### code_review/github.rs
- \`list()\` — PR list parsing with all status variants (open, merged, draft, etc.)
- \`details()\` — PR detail JSON parsing
- \`open_in_browser()\` — correct command arguments
- Pagination / empty results
- Malformed API response handling

### issue_tracker/github.rs
- \`list()\` — issue list with PR filtering (GitHub API includes PRs in issues)
- \`open_in_browser()\` — correct command arguments
- Empty results

## Difficulty

**Medium** — depends on GhApi trait. Straightforward mock responses.

### LINKED ISSUE #91: test: extend tmux.rs and zellij.rs workspace tests
## Context

Both \`tmux.rs\` (465 lines) and \`zellij.rs\` (491 lines) already have solid test suites covering:
- TOML serialization round-trip
- State path construction
- Pure helper functions (split_flag, append_command_args)
- Pruning logic

**Not yet tested**: The async \`WorkspaceManager\` trait methods that shell out to tmux/zellij.

### With CommandRunner (post #73)
- \`list()\` — parse tmux/zellij output into workspace panels
- \`create()\` — correct command sequence for new workspace
- \`focus()\` — correct focus commands
- Error handling when tool not available

### Without CommandRunner (can do now)
- State file migration / upgrade paths
- More edge cases in TOML parsing
- Pruning with complex workspace trees
- Tab counting edge cases

## Difficulty

**Easy** (extending existing pure tests) to **Medium** (async methods with CommandRunner).

### LINKED ISSUE #120: feat: record/replay test harness for daemon and TUI
## Summary

The daemon/TUI split via `DaemonHandle` and `CommandRunner` traits creates natural seams for record/replay testing:

- **Daemon side**: Record `CommandRunner` CLI outputs during a real session, replay them to test issue caching, correlation, refresh logic, etc. without hitting GitHub/git.
- **TUI side**: Record the `DaemonEvent` stream and `CommandResult` responses, replay them to test rendering, input handling, and state transitions without live providers.

Both sides get deterministic, reproducible test harnesses from the existing architecture.

## Approach

- Serialize events/commands/responses at the protocol boundary (already serde-compatible)
- Recording wrapper around `CommandRunner` / `DaemonHandle` to capture sessions
- Replay implementations that feed recorded data back
- Could also enable golden-file snapshot testing for UI output

### PR DESCRIPTION
**Pull request by rjwittams**
## Summary

- **Record/replay framework** (`replay.rs`) — multi-channel recording (CLI commands + GitHub API) with YAML fixtures, path masking for deterministic replay, and `RECORD=1` mode for regenerating fixtures from real interactions
- **Provider tests with real recording** — git (6 tests), wt (3 tests), GitHub code review (2 tests), GitHub issues (1 test), tmux (2 tests), zellij (2 tests)
- **Workspace lifecycle tests** — end-to-end create/switch/list verified via external commands through the recording runner

Closes #89, closes #90, closes #91, closes #120.

## Test plan

- [x] `cargo test --locked --workspace` — 448 tests pass
- [x] `cargo clippy --all-targets --locked -- -D warnings` — clean
- [x] `cargo fmt` — clean
- [x] `RECORD=1 cargo test -p flotilla-core` — all recording tests regenerate fixtures successfully
- [ ] Review fixture YAML files for reasonable recorded data

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/flotilla-core/src/providers/replay.rs [hunk 1]
```diff
@@ -0,0 +1,924 @@
...
+
+impl ReplayRunner {
+    pub fn new(session: ReplaySession) -> Self {
+        Self { session }
+    }
+}
+
+#[async_trait]
+impl CommandRunner for ReplayRunner {
+    async fn run(&self, cmd: &str, args: &[&str], _cwd: &Path) -> Result<String, String> {
```
[the remark this answers is not in the document]
Comment (rjwittams) [reply crates/flotilla-core/src/providers/replay.rs]: Fixed in 9a11bc3. ReplayRunner now asserts cwd matches the fixture value in both run() and run_output().

File: crates/flotilla-core/src/providers/workspace/zellij.rs [hunk 2]
```diff
@@ -477,4 +477,265 @@ mod tests {
...
+            env!("CARGO_MANIFEST_DIR"),
+            name
+        )
+    }
+
+    fn setup_zellij_ws_session() {
+        // Set ZELLIJ_SESSION_NAME BEFORE creating the runner so it inherits the env
+        unsafe {
+            std::env::set_var("ZELLIJ_SESSION_NAME", "flotilla-test-zj-ws");
+        }
```
[the remark this answers is not in the document]
Comment (rjwittams) [reply crates/flotilla-core/src/providers/workspace/zellij.rs]: The unsafe set_var is needed because ZellijWorkspaceManager::session_name() reads ZELLIJ_SESSION_NAME from the process env directly — can't use Command::env() since it's the provider that reads it, not a child process. The unsafe calls only run during manual RECORD=1 mode (guarded by `if recording { ... }`). The proper fix is making session_name() injectable via the struct, but that's a provider API refactoring beyond this PR's scope.

Separately, the parallel tmux session name collision is fixed in 0378f04 — setup_tmux_ws_session now uses a distinct name (flotilla-test-tmux-ws).
